roles/vault_operator/tasks/main.yaml

```diff
@@ -0,0 +1,130 @@
+    wait_condition:
+      type: Available
+      status: true
+
+- name: Apply RBAC manifests for Vault Operator
+  run_once: true
+  kubernetes.core.k8s:
+    state: present
+    definition:
+      - apiVersion: v1
+        kind: ServiceAccount
+        metadata:
+          name: vault
+          namespace: "{{ vault_operator_helm_release_namespace }}"
+
+      - apiVersion: rbac.authorization.k8s.io/v1
+        kind: Role
+        metadata:
+          name: vault
+          namespace: "{{ vault_operator_helm_release_namespace }}"
+        rules:
+          - apiGroups: [""]
+            resources: ["pods"]
+            verbs: ["get", "patch", "update"]
+          - apiGroups: [""]
+            resources: ["secrets"]
+            verbs: ["*"]
+
+      - apiVersion: rbac.authorization.k8s.io/v1
+        kind: RoleBinding
+        metadata:
+          name: vault
+          namespace: "{{ vault_operator_helm_release_namespace }}"
+        roleRef:
+          apiGroup: rbac.authorization.k8s.io
+          kind: Role
+          name: vault
+        subjects:
+          - kind: ServiceAccount
+            name: vault
+            namespace: "{{ vault_operator_helm_release_namespace }}"
+
+      - apiVersion: rbac.authorization.k8s.io/v1
+        kind: ClusterRoleBinding
+        metadata:
+          name: vault-auth-delegator
+          namespace: "{{ vault_operator_helm_release_namespace }}"
+        roleRef:
+          apiGroup: rbac.authorization.k8s.io
+          kind: ClusterRole
+          name: system:auth-delegator
+        subjects:
+          - kind: ServiceAccount
+            name: vault
+            namespace: "{{ vault_operator_helm_release_namespace }}"
+
+      - apiVersion: rbac.authorization.k8s.io/v1
+        kind: Role
+        metadata:
+          name: leader-election-role
+          namespace: "{{ vault_operator_helm_release_namespace }}"
+        rules:
+          - apiGroups: [""]
+            resources: ["configmaps"]
+            verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
+          - apiGroups: ["coordination.k8s.io"]
+            resources: ["leases"]
+            verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
+          - apiGroups: [""]
+            resources: ["events"]
+            verbs: ["create", "patch"]
+          - apiGroups: ["authentication.k8s.io"]
+            resources: ["tokenreviews"]
+            verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
+
+      - apiVersion: rbac.authorization.k8s.io/v1
+        kind: RoleBinding
+        metadata:
+          name: leader-election-rolebinding
+          namespace: "{{ vault_operator_helm_release_namespace }}"
+        roleRef:
+          apiGroup: rbac.authorization.k8s.io
+          kind: Role
+          name: leader-election-role
+        subjects:
+          - kind: ServiceAccount
+            name: vault
+            namespace: "{{ vault_operator_helm_release_namespace }}"
```

Comment:
Forgive me but why is this necessary?  Is this stuff that is missing from the upstream Helm chart?

Reply:
Yes it is. https://github.com/bank-vaults/vault-operator/blob/main/deploy/rbac/kustomization.yaml RBAC needs extra steps using kustomize (official documentation): https://bank-vaults.dev/docs/operator/ (step 2.). And all manifests are with hardcoded namespace and etc. I fixed this before: https://github.com/bank-vaults/vault-operator/pull/ 817